packages/server/src/fhir/operations/patientmerge.ts

```diff
@@ -51,28 +64,135 @@ const operation: OperationDefinition = {
+      documentation:
+        'The details of the Patient resource to use for the merge. Must have the same patient.id as target and include the link property. If not provided, the server will merge all identifiers from source and apply default merge logic.',
+      type: 'Patient',
+      min: 0,
+      max: '1',
+    },
+    {
+      use: 'in',
+      name: 'preview',
+      documentation:
+        'If true, returns what the merge result would be WITHOUT performing the merge. Useful for user review.',
+      type: 'boolean',
+      min: 0,
+      max: '1',
+    },
     {
       use: 'out',
-      name: 'return',
-      documentation: 'The updated target Patient resource',
+      name: 'input',
+      documentation: 'Echo of the input parameters',
+      type: 'Parameters',
+      min: 0,
+      max: '1',
+    },
+    {
+      use: 'out',
+      name: 'outcome',
+      documentation: 'OperationOutcome with information about the merge',
+      type: 'OperationOutcome',
+      min: 0,
+      max: '1',
+    },
+    {
+      use: 'out',
+      name: 'result',
+      documentation: 'The final merged target Patient resource',
       type: 'Patient',
-      min: 1,
+      min: 0,
       max: '1',
     },
     {
       use: 'out',
-      name: 'resourcesUpdated',
-      documentation: 'Number of clinical resources that had their references updated',
-      type: 'integer',
+      name: 'return',
+      documentation: 'Deprecated: use result instead. The updated target Patient resource',
+      type: 'Patient',
       min: 0,
       max: '1',
     },
   ] as OperationDefinitionParameter[],
 };
 
 export interface PatientMergeParameters {
-  'source-patient': Reference<Patient>;
+  'source-patient'?: Reference<Patient>;
+  'source-patient-identifier'?: Identifier | Identifier[];
   'target-patient'?: Reference<Patient>;
+  'target-patient-identifier'?: Identifier | Identifier[];
+  'result-patient'?: Patient;
+  preview?: boolean;
+}
+
+/**
+ * Resolves a patient from either a direct reference or identifier(s).
+ * @param repo - The repository to use for lookups.
+ * @param reference - Direct patient reference (if provided).
+ * @param identifiers - Identifier(s) to search by (if provided).
+ * @param paramName - Name of the parameter (for error messages).
+ * @returns The resolved patient reference.
+ * @throws OperationOutcomeError if resolution fails or finds multiple patients.
+ */
+async function resolvePatientReference(
+  repo: Repository,
+  reference: Reference<Patient> | undefined,
+  identifiers: Identifier | Identifier[] | undefined,
+  paramName: string
+): Promise<Reference<Patient>> {
+  // If direct reference provided, use it
+  if (reference) {
+    return reference;
+  }
+
+  // If identifiers provided, resolve them
+  if (identifiers) {
+    const idArray = Array.isArray(identifiers) ? identifiers : [identifiers];
+
+    // Build search query for identifiers
+    const searchParams = new URLSearchParams();
```

Comment:
I'm a bit surprised to see URLSearchParams used in this way. It looks like you are using it like a hash that can hold many values for the same key? This might be more cleanly expressed as something like a `[string, string][]`. Or maybe building up the Filter entries directly as you iterate?
```ts
const filters: Filter[] = identifiers
  .filter((identifier) => Boolean(identifier.value))
  .map((identifier) => ({
    code: 'identifier
    operator: 'eq' as const,
    value: identifier.system ? `${identifier.system}|${identifier.value}` : identifier.value
  })
const bundle = await repo.search({ resourceType: 'Patient', filters });
```